QuantConnect.ExanteBrokerage/ExanteBrokerage.cs

```diff
@@ -0,0 +1,850 @@
+            {
+                return;
+            }
+
+            var exanteSymbol = Client.GetSymbol(exanteOrder.OrderParameters.SymbolId);
+            var symbol = ConvertSymbol(exanteSymbol);
+
+            var security = _securityProvider.GetSecurity(symbol);
+            var orderFeeParameters = new OrderFeeParameters(security, order);
+            var fee = security.FeeModel.GetOrderFee(orderFeeParameters);
+
+            var orderEvent = new OrderEvent(order, DateTime.UtcNow, fee)
+            {
+                Status = ConvertOrderStatus(exanteOrder.OrderState.Status),
+            };
+            if (exanteOrder.OrderState.Status == ExanteOrderStatus.Filled)
+            {
+                orderEvent.FillQuantity = exanteOrder.OrderParameters.Quantity;
+                _messageHandler.WithLockedStream(() => { _orderMap.TryRemove(exanteOrder.OrderId, out _); });
+            }
+
+            OnOrderEvent(orderEvent);
+        }
+
+        private IEnumerable<BaseData> GetCandlesHistory(Data.HistoryRequest request)
+        {
+            var symbol = SymbolMapper.GetBrokerageSymbol(request.Symbol);
+
+            var exanteTickType = request.TickType switch
+            {
+                TickType.Quote => ExanteTickType.Quotes,
+                TickType.Trade => ExanteTickType.Trades,
+                TickType.OpenInterest => throw new ArgumentException($"Unsupported TickType requested {request.TickType}"),
+                _ => throw new ArgumentOutOfRangeException($"Unexpected TickType {request.TickType}")
+            };
+
+            var exanteTimeframe = request.Resolution switch
+            {
+                Resolution.Tick => throw new ArgumentException($"Unsupported Resolution requested {request.Resolution}"),
+                Resolution.Second => throw new ArgumentException($"Unsupported Resolution requested {request.Resolution}"),
+                Resolution.Minute => ExanteCandleTimeframe.Minute1,
+                Resolution.Hour => ExanteCandleTimeframe.Hour1,
+                Resolution.Daily => ExanteCandleTimeframe.Day1,
+                _ => throw new ArgumentOutOfRangeException($"Unexpected Resolution {request.Resolution}")
+            };
+
+            var history = Client.GetCandles(
+                symbol,
+                exanteTimeframe,
+                from: request.StartTimeUtc,
+                to: request.EndTimeUtc,
+                tickType: exanteTickType
+            ).Data.ToList();
+
+            var period = request.Resolution.ToTimeSpan();
+
+            foreach (var kline in history)
+            {
+                yield return request.TickType switch
+                {
+                    TickType.Quote =>
+                        new QuoteBar
+                        {
+                            Time = kline.Date,
+                            Symbol = request.Symbol,
+                            Value = kline.Close,
+                            DataType = MarketDataType.TradeBar,
```

Comment:
> DataType = MarketDataType.TradeBar,

Should use the QuoteBar and TradeBar constructors instead of the public setters, to avoid these issues.
Does exante provide bid/ask bar information for the quoteBar?